storage/replica_command.go

```diff
@@ -646,6 +650,13 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 func (r *Replica) HeartbeatTxn(batch engine.Engine, ms *engine.MVCCStats, args proto.HeartbeatTxnRequest) (proto.HeartbeatTxnResponse, error) {
 	var reply proto.HeartbeatTxnResponse
 
+	if args.Txn == nil {
```

Comment:
Yes, that was my understanding. I thought the check is nice to have to avoid potential bugs.